File: apps/java/FaceDetectorApp.java
import java.io.IOException;
import java.nio.file.Path;
import java.nio.file.Paths;

import org.sertiscorp.oneml.face.*;

public class FaceDetectorApp {

    public static void main(String[] args) throws IOException {
        Path basePath = Paths.get("", "../../assets/images");
        String path = basePath + "/face-detect-set/face/0.jpg";

        LicenseManager manager = new LicenseManager();
        manager.activateTrial();
        FaceDetector detector = new FaceDetector(manager);
        Utils utils = new Utils(manager);

        Image img = utils.readImageCV(path);
        FaceDetectorResult result = detector.detect(img);
        System.out.println("Faces: " + result.getSize());

        BBoxList boxes = result.getBBoxes();
        Landmark5List landmarks = result.getLandmarks();
        ScoreList scores = result.getScores();
        FacePoseList poses = result.getPoses();

        for (int i = 0; i < result.getSize(); i++) {
            System.out.println("Face " + i);
            System.out.println(String.format("Score: %.6f", scores.get(i)));
            System.out.println("Pose: " + poses.get(i));

            BBox box = boxes.get(i);
            String out = String.format("BBox: [(%.6f, %.6f), (%.6f, %.6f)]", box.getTop(), box.getLeft(), box.getBottom(), box.getRight());
            System.out.println(out);

            int n_landmarks = Face.getLandmarkSize();
            Landmark5 landmark = landmarks.get(i);
            for (int j = 0; j < n_landmarks; j++) {
                System.out.println(String.format("Landmark %d: (%.6f, %.6f)", j, landmark.getX().get(j), landmark.getY().get(j)));
            }
        }
    }
}

File: apps/csharp/FaceDetectorApp.cs
using OneML.Face;

class FaceDetectorApp {
    static void Main(string[] args) {
        LicenseManager manager = new LicenseManager();
        manager.ActivateTrial();

        using (FaceDetector detector = new FaceDetector(manager))
        using (Utils utils = new Utils(manager))
        {
            string basePath = @"../../assets/images/";
            string path = basePath + @"face-detect-set/face/0.jpg";
            Image image = utils.ReadImageCV(path);
            FaceDetectorResult result = detector.Detect(image);
            Console.WriteLine("Faces: " + result.GetSize());

            ScoreList scores = result.GetScores();
            BBoxList bboxes = result.GetBBoxes();
            Landmark5List landmarks = result.GetLandmarks();
            FacePoseList poses = result.GetPoses();
            StatusCodeList statuses = result.GetReturnStatus();

            string precision = "F6";
            for (int i = 0; i < result.GetSize(); i++) {
                Console.WriteLine("Face " + i);
                Console.WriteLine("status: " + statuses[i]);
                Console.WriteLine("Score: " + scores[i].ToString(precision));
                Console.WriteLine("Pose: " + poses[i]);

                BBox box = bboxes[i];
                string boxString = "BBox: [(" + box.top.ToString(precision) + ", " + box.left.ToString(precision) + ", " + box.bottom.ToString(precision) + ", " + box.right.ToString(precision) + ")]";
                Console.WriteLine(boxString);

                Landmark5 landmark = landmarks[i];
                for (int j = 0; j < Face.GetLandmarkSize(); j++) {
                    Console.WriteLine("Landmark " + j + ": (" + landmark.x[j].ToString(precision) + ", " + landmark.y[j].ToString(precision) + ")");
                }
            }

            UsageReport report = detector.GetUsage();
            report.ToLog();
        }
    }
}